src/Domain/TheCocktailDb/Service/Client.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\TheCocktailDb\Service;
+
+use App\Application\IngredientCollection\IngredientCollectionDenormalizer;
+use App\Application\IngredientCollection\IngredientDenormalizer;
+use App\Domain\Dto\IngredientCollection;
+use App\Domain\Ingredients\Dto\Ingredient;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactory;
+use Symfony\Component\Serializer\Mapping\Loader\AnnotationLoader;
+use Symfony\Component\Serializer\NameConverter\MetadataAwareNameConverter;
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Doctrine\Common\Annotations\AnnotationReader;
+
+class Client
+{
+    private const GET_RANDOM_MEAL_ENDPOINT = '/1/random.php';
+    private const GET = 'GET';
+    private HttpClientInterface $http;
+
+    public function __construct()
+    {
+        $this->http = HttpClient::create();
+    }
+
+    public function getCocktail()
+    {
+         $response = $this->http->request(
+             self::GET,
+             'https://www.thecocktaildb.com/api/json/v1' . self::GET_RANDOM_MEAL_ENDPOINT
+         ); // TODO: maybe transfer API request information to yaml
+
+        $content = $response->getContent();
+
+        // TODO: extract further logic to new class
+        $classMetadataFactory = new ClassMetaDataFactory(new AnnotationLoader(new AnnotationReader()));
+        $metadataAwareNameConverter = new MetadataAwareNameConverter($classMetadataFactory);
+        $encoders = [new JsonEncoder()];
+        $normalizers = [new ObjectNormalizer($classMetadataFactory, $metadataAwareNameConverter)];
+        $denormalizer = new IngredientCollectionDenormalizer(new IngredientDenormalizer());
+        $serializer = new Serializer($normalizers, $encoders);
```

Comment:
You reaaaally need dependency injection for this serializer :D

Reply:
Implemented with de8d9a3 commit